src/core/component.js

```diff
@@ -203,6 +204,7 @@ Component.prototype = {
       // If value is an object, copy it to our pooled newAttrValue object to use to update
       // the attrValue.
       tempObject = this.objectPool.use();
+      utils.objectPool.clearObject(tempObject);
```

Comment:
So I understand. `use` function should already calls [clearObject](https://github.com/aframevr/aframe/blob/master/src/utils/object-pool.js#L26). Is not this redundant?

Reply:
Oops, seems when cleaning up the PR I got `clearObject` and `removeUnusedKeys` mixed up in this place. The object pool does indeed "clear" the object, but the keys remains. Looking at the code again in more detail, it is actually safe here to have superfluous keys on `tempObject` in the subsequent `utils.extend` and `extendProperties` calls.